Sources/XiCliCore/CommandLineTool.swift

```diff
@@ -0,0 +1,155 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Cocoa
+
+public final class CommandLineTool {
+    private let arguments: [String]
+    
+    public init(arguments: [String] = CommandLine.arguments) {
+        self.arguments = Array(arguments.dropFirst())
+    }
+    
+    public func run() throws {
+        if arguments.isEmpty {
+            help()
+            return
+        }
+        
+        let args = try getArguments(from: arguments)
+        
+        if let args = args {
+            let filePath = try resolvePath(from: args.fileInput)
+            try openFile(at: filePath)
+            
+            if args.wait {
+                print("waiting for editor to close...")
+                let group = DispatchGroup()
+                group.enter()
+                setObserver(group: group, filePath: filePath)
+                group.wait()
+                print("closed")
+            }
+        } else {
+            help()
+        }
+    }
+    
+    func getArguments(from args: [String]) throws -> Arguments? {
+        var fileName: String?
+        var wait = false
+        for arg in args {
+            if arg == "--wait" || arg == "-W" {
+                wait = true
+            } else if arg == "--help" || arg == "-h" {
+                return nil
+            } else {
+                fileName = arg
+            }
+        }
+        
+        guard let comfFileName = fileName else {
+            throw Error.missingFileName
+        }
+        
+        return Arguments(fileInput: comfFileName, wait: wait)
+    }
+    
+    func resolvePath(from input: String) throws -> String {
+        let fileManager = FileManager.default
+        var filePath: URL!
+        
+        if input.hasPrefix("/") {
+            filePath = URL(fileURLWithPath: input)
+        } else if input.hasPrefix("~") {
+            var input = input
+            input.remove(at: input.startIndex)
+            if #available(OSX 10.12, *) {
+                filePath = fileManager.homeDirectoryForCurrentUser
+            } else {
+                filePath = URL(fileURLWithPath: NSHomeDirectory())
+            }
+            filePath.appendPathComponent(input)
+            print(filePath)
+        } else {
+            let basePath = URL(fileURLWithPath: fileManager.currentDirectoryPath)
+            filePath = basePath.appendingPathComponent(input)
+        }
+        
+        let pathString = filePath.path
+        
+        if !fileManager.fileExists(atPath: pathString) {
+            print("creating \(pathString)...")
+            let createSuccess = fileManager.createFile(atPath: pathString, contents: nil, attributes: nil)
+            
+            guard createSuccess else {
+                throw Error.couldNotCreateFile
+            }
+        }
+        
+        return pathString
+    }
+    
+    func openFile(at path: String) throws {
+        print("opening \(path)...")
```

Comment:
ditto, if it opens we know about it and it if fails we'll show an error.

Reply:
Removed!